src/framework/analyses.ml

```diff
@@ -274,13 +274,13 @@ struct
         iterGlobals file (function
             | GFun (_, loc)
             | GVar (_, _, loc) ->
-              SH.replace files loc.file (Hashtbl.find_option Preprocessor.dependencies loc.file)
+              SH.replace files loc.file (Hashtbl.find_default Preprocessor.dependencies loc.file [])
```

Comment:
Why this change if the data structure is the same?

Reply:
I personally think having `null` instead of the empty list in the JSON is ugly, but I can switch it back if we want to do that.

Reply:
Previously the possibility of `None`/`null` was necessary for the cases where we failed to extract the dependencies via `-MD`. I guess now they should always be there, so it should be safe to just use `Hashtbl.find` without the default here. That way if there's ever any issue, it's immediately obvious via the exception rather than being silently replaced by pretending to have no dependencies, which could even be unsound.

Reply:
No, this dependencies map can still be empty for header files that are not seen by the cpp on its own. 
In my example this is the case for  `/usr/include/x86_64-linux-gnu/bits/byteswap.h`.

It may have dependencies, and they are transitively included in any file passed to the CPP, but not recorded for the file itself.

So maybe you're right and we should go back to having `None/null` here.

Reply:
That `/usr/include/x86_64-linux-gnu/bits/byteswap.h` shouldn't appear in the dependency keys in the first place. If it's included by something, it's already listed as a dependency of a C file, so a header on its own makes zero sense. I'm not exactly sure where it now suddenly appears from, but my suggestion here https://github.com/goblint/analyzer/pull/587#issuecomment-1029102095 would actually work around this problem altogether.

Reply:
It is not in the keys, but is looked up because some `GVar`s have it as their location.

Reply:
As mentioned by the TODO there, it's probably incomplete as an approach anyway. The complete approach would be to just dump the entire dependencies structure, that must necessarily contain all filenames that could possibly occur everywhere. And it would be much simpler than having a CIL visitor go through the whole AST just to find all the filenames, which the dependencies structure already has.